core/src/main/scala/kafka/server/metadata/BrokerMetadataSnapshotter.scala

```diff
@@ -45,26 +45,37 @@ class BrokerMetadataSnapshotter(
    */
   private var _currentSnapshotOffset = -1L
 
+  /**
+   * The offset of the newest snapshot, or -1 if there hasn't been one. Accessed only under
+   * the object lock.
+   */
+  private var _latestSnapshotOffset = -1L
```

Comment:
Do we need to initialize this on startup? If we don't, is it possible that we would end up snapshotting the same offset more than once?

Reply:
This seems an existing problem, I checked the logic of `QuorumController.SnapshotGeneratorManager`, it will check the result of `raftClient.createSnapshot` and just skip it if it's empty, however, at broker side, `BrokerSnapshotWriterBuilder` will throw an exception on empty snapshot. how about making it consistent with controller code.